dali/operators/generic/permute_batch.h

```diff
@@ -45,9 +45,10 @@ class PermuteBatchBase : public Operator<Backend> {
       this->spec_.TryGetRepeatedArgument(indices_, "indices");
     }
     // TODO(michalz): Remove when fully variable batch size is supported
-    DALI_ENFORCE(static_cast<int>(indices_.size()) == ws.GetRequestedBatchSize(0), make_string(
+    int cur_batch_size = InferBatchSizeFromInput(ws);
```

Comment:
It accepts a regular input, so the check will enforce the same input size as it did in non `split_` case (although the same thing is checked at the op level), but enables using it within split_ branches,